### PR TITLE
fix: wrong optional dependencies in complex demo

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

File: test/fixtures/plugin-complex-dependencies/plugin/antvip/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "eggPlugin": {
+    "name": "antvip",
+    "dependencies": [
```
Comment (leoner) [inline test/fixtures/plugin-complex-dependencies/plugin/antvip/package.json]: antvip 也依赖 tr

File: test/fixtures/plugin-complex-dependencies/plugin/drm/package.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "eggPlugin": {
+    "name": "drm",
+    "dependencies": [
```
Comment (leoner) [inline test/fixtures/plugin-complex-dependencies/plugin/drm/package.json]: 也依赖 tr

File: test/fixtures/plugin-complex-dependencies/config/plugin.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const path = require('path');
+
+//
+// tr -> a
+// ldc > b
+// zoneclient -> c
```
Comment (fengmk2) [inline test/fixtures/plugin-complex-dependencies/config/plugin.js]: @popomore 这些名词还是改成其他，不要暴露。。。

Comment (popomore) [inline test/fixtures/plugin-complex-dependencies/config/plugin.js]: 这个暴露问题不大吧，也看不出什么问题。

File: lib/utils/sequencify.js [hunk 4]
```diff
@@ -25,7 +30,6 @@ function sequence(tasks, names, results, missing, recursive, nest, optional) {
       }
       nest.pop(name);
     }
-    if (optional === true) return;
```
Comment (popomore) [inline lib/utils/sequencify.js]: 之前的修复有问题 https://github.com/eggjs/egg-core/pull/76

File: test/loader/mixin/load_plugin.test.js [hunk 5]
```diff
@@ -471,4 +471,26 @@ describe('test/load_plugin.test.js', function() {
     assert(warn.callCount === 0);
   });
 
+  it.only('should parse complex dependencies', () => {
```
Comment (dead-horse) [inline test/loader/mixin/load_plugin.test.js]: only

**Comment by fengmk2** [comment]
3.5.0

**Pull request merged**